### PR TITLE
fix: ESM tests on Windows.

### PR DESCRIPTION
**Pull request by liuxingbaoyu**
| Q                        | A 
| ------------------------ | ---
| License                  | MIT

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/14804"></a>

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -12,10 +12,22 @@ module.exports = {
...
 
   collectCoverageFrom: [
-    "packages/*/src/**/*.{js,mjs,ts}",
-    "codemods/*/src/**/*.{js,mjs,ts}",
-    "eslint/*/src/**/*.{js,mjs,ts}",
+    "packages/*/src/**/*.{js,cjs,mjs,ts}",
+    "codemods/*/src/**/*.{js,cjs,mjs,ts}",
+    "eslint/*/src/**/*.{js,cjs,mjs,ts}",
   ],
+  // coveragePathIgnorePatterns Doesn't work
```
Comment (liuxingbaoyu) [inline jest.config.js:19]: This is probably a jest bug, I can't find a way to make it work, even using `collectCoverageFrom:["!packages/babel-core/src/vendor/**"]`

Comment (nicolo-ribaudo) [reply jest.config.js:19]: Maybe we need something like this to handle windows paths?
```js
coveragePathIgnorePatterns: [
  /packages[\\/]babel-core[\\/]src[\\/]vendor[\\/].*/
]
```

Comment (liuxingbaoyu) [reply jest.config.js:19]: No, jest only supports regex strings. Even if I use ".*", it still doesn't work, so I guess it's a bug, there's been a similar issue in the jest repo for several years.

**Review by JLHwung — APPROVED** [review]
This PR looks good to me. But I don't have access to a working Windows environment. Can we add Windows ESM tests to CI?

**Comment by liuxingbaoyu** [comment]
https://github.com/babel/babel/runs/7798363149?check_suite_focus=true

This seems to have been included in `CI` and I can see skip messages.

File: jest.config.js (hunk 1 above)
Comment (nicolo-ribaudo) [inline jest.config.js:19]:
```suggestion
  // coveragePathIgnorePatterns Doesn't work on windows
```

**Review by nicolo-ribaudo — APPROVED** [review]

**Pull request merged**